docs/source/scylla_cluster_autoscaler_crd.md

```diff
@@ -0,0 +1,104 @@
+# Scylla Cluster Autoscaler CRD
+
+Scylla clusters can be created and configuring using the 'scyllaclusterautoscalers.scylla.scylladb.com' custom resource definition (CRD).
+
+Please refer to the the [user guide walk-through](generic.md) for deployment instructions.
+This page will explain all the available configuration options on the Scylla Cluster Autoscaler CRD.
+
+## Sample
+
+```yaml
+apiVersion: scylla.scylladb.com/v1alpha1
+kind: ScyllaClusterAutoscaler
+metadata:
+  name: example-sca
+spec:
+  targetRef:
+    namespace: scylla
+    name: example-cluster
+  updatePolicy:
+    updateMode: Auto
+  scalingPolicy:
+    datacenters:
+    - name: us-east-1
+      racks:
+      - name: us-east-1a
+        rules:
+        - name: cpu utilization horizontal up
+          priority: 1
+          expression: 'avg(scylla_reactor_utilization{scylla_cluster="example-cluster", scylla_datacenter="us-east-1", scylla_rack="us-east-1a"}) > bool 70'
+          mode: Horizontal
+          for: 10m
+          step: 30s
+          factor: 2
+        - name: cpu utilization horizontal down
+          priority: 1
+          expression: 'avg(scylla_reactor_utilization{scylla_cluster="example-cluster", scylla_datacenter="us-east-1", scylla_rack="us-east-1a"}) < bool 10'
+          mode: Horizontal
+          for: 10m
+          step: 30s
+          factor: 0.5
+        memberPolicy:
+          minAllowed: 1
+          maxAllowed: 5
+        resourcePolicy:
+          minAllowedCpu: 1
+          maxAllowedCpu: 2
+          controlledValues: Requests
+status:
+  lastApplied: 2021-04-14T11:54:17Z
+  lastUpdated: 2021-04-14T16:43:22Z
+  updateStatus: Ok
+  recommendations:
+    datacenterRecommendations:
+      - name: us-east-1
+        rackRecommendations:
+          - name: us-east-1a
+            members: 2
+            resources:
+              limits:
+                cpu:     2
+                memory:  30Gi
+              requests:
+                cpu:     2
+                memory:  30Gi
+```
+
+## Autoscaler Settings
+
+* `targetRef`: description if the ScyllaCluster object the autoscaling configuration is referring to. Comprising of `name` and `namespace`.
+  * `namespace`: namespace of ScyllaCluster
+  * `name`: name of ScyllaCluster
+
+* `updatePolicy`: Optional field. Rules and limitations of how the target is meant to be updated.
+  * `updateMode`: Optional field. Can be set to either "Off" or "Auto" (default "Auto"). Recommendations are being provided and saved in either of these cases, however, they are only ever applied in the latter.
+  * `recommendationExpirationTime`: Optional field. How long the recommendations stay valid.
```

Comment:
Some of the fields have default values. Mention them at the end of the description.